tests/unit/test.program.js

```diff
@@ -430,6 +433,203 @@ describe('program.main', () => {
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch((error) => {
+        if (!(error instanceof UsageError)) {
+          assert.match(error.message, /some error/);
+          sinon.assert.notCalled(fakeApplyConfigToArgv);
+          sinon.assert.notCalled(fakeProcess.exit);
+          sinon.assert.calledOnce(logger.error);
+        }
+      });
+  });
+
+  it('logs and throws when fakeApplyConfigToArgv throws UsageError', () => {
+    const fakeProcess = createFakeProcess();
+    const logger = fake(createLogger(__filename));
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {
+      throw new UsageError('bad config option');
+    });
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(UsageError, (error) => {
+        assert.match(error.message, /bad config option/);
+        sinon.assert.notCalled(fakeProcess.exit);
+        sinon.assert.calledOnce(logger.error);
+      }));
+  });
+
+  it('logs and throws when fakeApplyConfigToArgv throws WebExtError', () => {
+    const fakeProcess = createFakeProcess();
+    const logger = fake(createLogger(__filename));
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {
+      throw new WebExtError('bad config option');
+    });
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(WebExtError, (error) => {
+        assert.match(error.message, /bad config option/);
+        sinon.assert.notCalled(fakeProcess.exit);
+        sinon.assert.calledOnce(logger.error);
+      }));
+  });
+
+  it('logs and throws when fakeApplyConfigToArgv throws WebExtError', () => {
```

Comment:
This seems to be testing mostly the same scenario of the test above it, but it should probably be the test case which explicitly test what is going to happen if "fakeApplyConfigToArgv throws an unexpected error", am I right?